File: Java/Csharp/TemplateMethod/TemplateMethod/Class2.cs
using System;

namespace TemplateMethod
{
    class Class2 : AbstractClass
    {
        protected override void RequiredOperations1()
        {
            Console.WriteLine("Class2 says: Implemented Operation1");
        }

        protected override void RequiredOperation2()
        {
            Console.WriteLine("Class2 says: Implemented Operation2");
        }

        protected override void Hook1()
        {
            Console.WriteLine("Class2 says: Overridden Hook1");
        }
    }
}

File: TheRayCode/PHP/Behavioral/Template/Example/Class2.php
<?php

class Class2 extends AbstractClass
{
    protected function requiredOperations1(): void
    {
        echo "ConcreteClass2 says: Implemented Operation1<br/>";
    }

    protected function requiredOperation2(): void
    {
        echo "ConcreteClass2 says: Implemented Operation2<br/>";
    }

    protected function hook1(): void
    {
        echo "ConcreteClass2 says: Overridden Hook1<br/>";
    }
}